src/App.jsx

```diff
@@ -0,0 +1,294 @@
+
+  // effects
+  useEffect(() => {
+    if (state.gameStatus.running && isBootleTurn) {
+      onClickRollDiceHandler()
+      setIsBottleTurn(false)
+    }
+    return () => {
+      if (debugTimeInterval) {
+        log(`useEffect: cleanUp interval ${rollInterval}`)
+      }
+    }
+  }, [state, isBootleTurn])
+
+  // eventHandlers
+  const onClickStartGameHandler = () => {
+    setUIDisabled(true)
+    dispatch({
+      type: 'START_GAME'
+    })
+    // give some time for bottle think its play and game state finish setup
+    setTimeout(() => {
+      setIsBottleTurn(true)
+    }, constants.BOTTLE_THINK_INTERVAL)
+  }
+
+  const onClickStopGameHandler = () =>
+    dispatch({
+      type: 'STOP_GAME'
+    })
+
+  const onClickRollDiceHandler = () => {
+    const debug = false
+    setUIDisabled(true)
+    const roleDirectionStatus = randomNumber(1, 6)
+    const roleStepsStatus = randomNumber(1, 6)
+    const directionStatus = getDirection(
+      state,
+      state.gameStatus.directionStatus,
+      roleDirectionStatus,
+      roleStepsStatus
+    )
+    if (debug) {
+      log(
+        `roleDirectionStatus: ${roleDirectionStatus}, roleStepsStatus: ${roleStepsStatus}, directionStatus: ${directionStatus}`
+      )
+    }
+
+    // start rolling
+    dispatch({
+      type: 'ROLL_DICE',
+      payload: {
+        directionStatus,
+        roleDirectionStatus,
+        roleStepsStatus
+      }
+    })
+
+    let moveTo
+    let dispatchType
+    if (state.gameStatus.turn === constants.PLAYER_ID) {
+      moveTo = state.gameStatus.playerStatus.position
+      dispatchType = 'MOVE_PLAYER'
+    } else {
+      moveTo = state.gameStatus.bottleStatus.position
+      dispatchType = 'MOVE_BOTTLE'
+    }
+
+    // move steps
+    let steps = 0
+    const interval = setInterval(() => {
+      if (debugTimeInterval) {
+        log(`clearInterval tick ${rollInterval}`)
+      }
+      if (debug) {
+        log(`steps:${steps} === roleStepsStatus:${roleStepsStatus}`)
+      }
+      if (steps === roleStepsStatus - 1) {
+        dispatch({
+          type: 'CHANGE_TURN',
+          payload: {
+            directionStatus,
+            roleDirectionStatus,
+            roleStepsStatus
+          }
+        })
+        clearInterval(interval)
+        if (debugTimeInterval) {
+          log(`clearInterval interval ${rollInterval}`)
+        }
+        // pass turn to bottle
+        if (state.gameStatus.turn === constants.PLAYER_ID) {
+          setIsBottleTurn(true)
+        } else {
+          setUIDisabled(false)
+        }
+      }
+
+      // used as next position
+      moveTo = getMoveStepPositions(
+        moveTo,
+        directionStatus,
+        roleStepsStatus,
+        state.gameStatus.columns,
+        state.gameStatus.rows
+      )
+
+      // dispatch moveTo
+      dispatch({
+        type: dispatchType,
+        payload: {
+          position: moveTo
+        }
+      })
+
+      // check player positions to define end of game
```

Comment:
it is better to split the logic of moving, victory checking, etc. into different functions or methods. 
Also, it is better to keep it in a separate file, not to overload this component and this handler

Reply:
FIXED extracted two functions from onClickRollDiceHandler
rollDice() and checkEndGame()

onClickRollDiceHandler is a lot more clean

```js
  const onClickRollDiceHandler = () => {
    rollDice(state, dispatch, setUIDisabled, rollInterval)
  }
```